sdk/src/Core/Amazon.Runtime/Credentials/Internal/_bcl45+netstandard/SsoTokenUtils.cs

```diff
@@ -92,6 +92,21 @@ public static string AsJson(this SsoToken token)
             return ToJson(token);
         }
 
+        /// <summary>
+        /// This determines whether the <seealso cref="SsoToken.RegistrationExpiresAt"/> field is 5 minutes within expiration
+        /// </summary>
+        /// <param name="token">The sso token</param>
+        /// <returns>This returns true if <seealso cref="SsoToken.RegistrationExpiresAt"/> is within 5 minutes of expiration. False otherwise</returns>
+        public static bool RegisteredClientExpired(this SsoToken token)
+        {
+            if (null == token)
+                throw new ArgumentNullException(nameof(token));
+            DateTime dateTime = ConvertRFC3339StringToDateTime(token.RegistrationExpiresAt);
```

Comment:
@peterrsongg In order to remain compatible with existing cached tokens, or tokens written out by other systems, tokens will be loaded that contain null values for the RegistrationExpiresAt field. In these cases, can we avoid the exception that is thrown in ConvertRFC3339StringToDateTime by early-exiting before this line?

The RegistrationExpiresAt field is optional, so if RegistrationExpiresAt is null, we have two options:

- we can return false and assume the registered client hasn't expired. An attempt to refresh the token using the SDK changes in this PR would succeed if the client registration hasn't actually expired (only the backend knows at this point). At some point, token refreshes may start to fail due to client registration expiry. I believe this enters into the state that we're trying to fix with this change.
- we can return true and assume the registered client has expired. This would cause the code to request an all-new token (if options.SupportsGettingNewToken is true), and in doing so, will register a new device, and update the registration expires at

I recommend option two, but believe that either handling will be an improvement over the exception.